app/controllers/static_pages_controller.rb

```diff
@@ -1,5 +1,9 @@
 class StaticPagesController < ApplicationController
-  def home; end
+  def home
+    return unless logged_in?
```

Comment:
có hàm `logged_in_user` rồi kìa e

Reply:
theo thiết kế của sách thì hàm logged_in_user dùng để kiểm tra user có được login hay chưa, nếu chưa thì sẽ báo lỗi qua flash và hàm logged_in_user sử dụng hàm logged_in? để kiểm tra. Còn ở đây trang này chỉ kiểm tra log in hay chưa để hiển thị trang theo 2 trường hợp khác nhau, nếu sử dụng logged_in_user ở đây thì khi người dùng không đăng nhập mà truy cập vào trang thì trang sẽ cứ báo lỗi qua flash. Em nghĩ có thể tên 2 hàm này hơi nhập nhằng thôi ạ còn về mặt thiết kế em nghĩ là sách làm hợp lý.

Reply:
`return unless logged_in?` line này thừa á em

Reply:
ơ sao thừa chị, ví dụ chưa login thì return luôn chớ không xuống dưới chạy current_user ra nil xong gọi .microposts là bị văng ra exception sao chị ?

Reply:
ukm, thừa lắm...
nhưng chị note thiếu ahihi
dùng callback ri em nè `before_action :logged_in_user, only : : home`
Có code rồi ko dùng, mà code thêm lần nữa chi vại

Reply:
![image](https://user-images.githubusercontent.com/18006732/41269540-ce964532-6e2f-11e8-8ac8-8181c0f468da.png)
làm như chị nói là hắn ra như ri nè chị -_- mỗi lần vào home nó sẽ luôn đòi login do cái hàm logged_in_user nó viết như vậy rồi, nếu sửa lại hàm logged_in_user thì nó sẽ lại lỗi chỗ khác, còn chỗ đó chỉ cần check login hay chưa để xem có load micropost hay không thôi mà chị